### ISSUE TITLE
 use <script setup> get the parent component

### ISSUE THREAD

**Issue by wb670** [quasarframework/quasar#9880]
/quasar/src/utils/private/vm.js

when use <script setup> 
the child component get the parent component,  proxy.$parent is {}
must use proxy.$.parent.proxy get parent component

so i added && typeof vm.$parent !== 'object'

```
export function getParentVm (vm) {
  if (vm.$parent !== void 0 && vm.$parent !== null && typeof vm.$parent !== 'object') {
    return vm.$parent
  }

  vm = vm.$.parent

  while (vm !== void 0 && vm !== null) {
    if (vm.proxy !== void 0 && vm.proxy !== null) {
      return vm.proxy
    }

    vm = vm.parent
  }
}
```

or

```
import { useContext,getCurrentInstance } from 'vue'
const ctx = useContext()
@ts-ignore
const { proxy }  = getCurrentInstance()
ctx.expose(proxy)
```

**Comment by rstoenescu** [comment]
Hi @wb670 ,

Can you create a reproduction repo pls? Trying to understand what you are doing.

**Comment by rstoenescu** [comment]
Enhancement will be available in Quasar v2.3.0.
